### ISSUE TITLE
Support SASL

### ISSUE THREAD

**Issue by ipmb** [BotBotMe/botbot-bot#46]
Freenode is now requiring SASL authentication for some nodes. 

Here's another IRC bot in Go that says it has functional SASL authentication: https://github.com/whyrusleeping/hellabot/blob/f58c1728b34404eb60c20f5e1cc05f70b7820ea1/hellabot.go#L141-L166

We would need to plug something like that in here: https://github.com/BotBotMe/botbot-bot/blob/7649e9c92c9275a9b9d3a79d1e7f490209ce53ab/network/irc/irc.go#L524-L532

This supersedes https://github.com/BotBotMe/botbot-web/issues/165

**Comment by ipmb** [comment]
```
I0412 13:04:42.500030   10904 irc.go:570] [RAW [o__o] on chat.freenode.net:6697 (0xc2082228f0) ] <-- :weber.freenode.net NOTICE * :*** No Id
ent response
I0412 13:04:42.500209   10904 irc.go:570] [RAW [o__o] on chat.freenode.net:6697 (0xc2082228f0) ] <-- :weber.freenode.net 451 * :You have not
 registered
I0412 13:04:42.500594   10904 irc.go:570] [RAW [o__o] on chat.freenode.net:6697 (0xc2082228f0) ] <-- :weber.freenode.net NOTICE [o__o] :*** Notice -- You need to identify via SASL to use this server
I0412 13:04:42.502484   10904 irc.go:570] [RAW [o__o] on chat.freenode.net:6697 (0xc2082228f0) ] <-- ERROR :Closing Link: ec2-54-85-45-223.compute-1.amazonaws.com (SASL access only)
```

**Comment by ipmb** [comment]
Additional info https://github.com/charybdis-ircd/charybdis/blob/release/3.5/doc/sasl.txt

**Comment by ipmb** [comment]
Resolved in #47
